src/Microsoft.AspNetCore.Mvc.Core/Internal/FileStreamResultExecutor.cs

```diff
@@ -18,18 +23,72 @@ public FileStreamResultExecutor(ILoggerFactory loggerFactory)
 
         public Task ExecuteAsync(ActionContext context, FileStreamResult result)
         {
-            SetHeadersAndLog(context, result);
-            return WriteFileAsync(context, result);
+            var fileLength = 0L;
```

Comment:
For non-seekable streams, calls to Length throw an exception. So I cannot get the length of the file. What's the expected behavior here? @Tratcher